.bazelrc

```diff
@@ -0,0 +1,182 @@
+build:msan --copt=-O0
+build:msan --copt=-fsanitize-memory-track-origins
+build:msan --copt=-fsanitize-memory-use-after-dtor
+build:msan --copt=-fno-omit-frame-pointer
+build:msan --copt=-DGPR_NO_DIRECT_SYSCALLS
+build:msan --linkopt=-fsanitize=memory
+build:msan --action_env=MSAN_OPTIONS=poison_in_dtor=1
+
+build:tsan --strip=never
+build:tsan --copt=-fsanitize=thread
+build:tsan --copt=-fno-omit-frame-pointer
+build:tsan --copt=-DGPR_NO_DIRECT_SYSCALLS
+build:tsan --copt=-DGRPC_TSAN
+build:tsan --linkopt=-fsanitize=thread
+build:tsan --action_env=TSAN_OPTIONS=suppressions=test/core/test_util/tsan_suppressions.txt:halt_on_error=1:second_deadlock_stack=1
+
+# a TSAN build that tries to create new threads whenever possible
+build:thready_tsan --strip=never
+build:thready_tsan --copt=-fsanitize=thread
+build:thready_tsan --copt=-fno-omit-frame-pointer
+build:thready_tsan --copt=-DGPR_NO_DIRECT_SYSCALLS
+build:thready_tsan --copt=-DGRPC_TSAN
+build:thready_tsan --copt=-DGRPC_MAXIMIZE_THREADYNESS
+build:thready_tsan --linkopt=-fsanitize=thread
+build:thready_tsan --action_env=TSAN_OPTIONS=suppressions=test/core/test_util/tsan_suppressions.txt:halt_on_error=1:second_deadlock_stack=1
+
+build:tsan_macos --strip=never
+build:tsan_macos --copt=-fsanitize=thread
+build:tsan_macos --copt=-fno-omit-frame-pointer
+build:tsan_macos --copt=-DGPR_NO_DIRECT_SYSCALLS
+build:tsan_macos --copt=-DGRPC_TSAN
+build:tsan_macos --linkopt=-fsanitize=thread
+build:tsan_macos --action_env=TSAN_OPTIONS=suppressions=test/core/test_util/tsan_suppressions.txt:halt_on_error=1:second_deadlock_stack=1
+build:tsan_macos --dynamic_mode=off
+
+build:ubsan --strip=never
+build:ubsan --copt=-fsanitize=undefined
+build:ubsan --copt=-fsanitize-link-c++-runtime
+build:ubsan --copt=-fno-omit-frame-pointer
+build:ubsan --copt=-DGRPC_UBSAN
+build:ubsan --copt=-DNDEBUG
+build:ubsan --copt=-fno-sanitize=function,vptr
+build:ubsan --linkopt=-fsanitize=undefined
+# avoid ubsan build error with int128 by linking against libc++
+# see b/200667821
+build:ubsan --linkopt=-fsanitize-link-c++-runtime
+build:ubsan --linkopt=-lc++
+build:ubsan --linkopt=-lc++abi
+build:ubsan --linkopt=-lm
+build:ubsan --action_env=UBSAN_OPTIONS=halt_on_error=1:print_stacktrace=1:suppressions=test/core/test_util/ubsan_suppressions.txt
+# if you want to run ubsan locally, uncomment following three lines
+# build:ubsan --linkopt=-lgcc_s
+# build:ubsan --linkopt=-lubsan
+# build:ubsan --linkopt=--rtlib=compiler-rt
+
+build:python_single_threaded_unary_stream --test_env="GRPC_SINGLE_THREADED_UNARY_STREAM=true"
+
+build:python_poller_engine --test_env="GRPC_ASYNCIO_ENGINE=poller"
+
+# Compile database generation config
+build:compdb --build_tag_filters=-nocompdb --features=-layering_check
+
+try-import %workspace%/tools/fuzztest.bazelrc
+build:fuzztest --cxxopt=-std=c++17
+
+build:fuzztest_test --cxxopt=-std=c++17
+
+# OpenTelemetry C++ needs this option to build with absl
+build --@io_opentelemetry_cpp//api:with_abseil
```

Comment:
I'm a bazel noob - but if this is specific to the codelab, can we keep these files (.bazelrc, and WORKSPACE) in the codelab directly. and cp them to other cpp codelabs as required?

Reply:
done